### PR TITLE
Add hydra-core recipe

### PR DESCRIPTION
**Pull request by mdraw**
https://hydra.cc/

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mdraw** [comment]
1. Can anyone help with the Windows pytest failure? I don't understand the error (https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/108950/logs/37).
2. I'm ignoring the `conf_loader` test here because it fails locally on my machine. Any idea why this could be happening? The traceback here is:
```
=================================== FAILURES ===================================
_____ TestConfigLoader.test_load_yml_file[pkg://hydra.test_utils.configs] ______

self = <tests.test_conf_loader.TestConfigLoader object at 0x7fc1e29db730>
path = 'pkg://hydra.test_utils.configs'

    def test_load_yml_file(self, path):
        config_loader = ConfigLoader(config_search_path=create_search_path([path]))
>       cfg = config_loader.load_configuration(config_file="config.yml", strict=False)

tests/test_conf_loader.py:148: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <hydra._internal.config_loader.ConfigLoader object at 0x7fc1e28b8940>
config_file = 'config.yml', overrides = [], strict = False

    def load_configuration(self, config_file, overrides=[], strict=None):
        assert config_file is None or isinstance(config_file, str)
        assert strict is None or isinstance(strict, bool)
        assert isinstance(overrides, list)
        if strict is None:
            strict = self.default_strict
    
        assert overrides is None or isinstance(overrides, list)
        overrides = copy.deepcopy(overrides) or []
    
        if config_file is not None and not self.exists_in_search_path(config_file):
>           raise MissingConfigException(
                missing_cfg_file=config_file,
                message="Cannot find primary config file: {}\nSearch path:\n{}".format(
                    config_file,
                    "\n".join(
                        [
                            "\t{} (from {})".format(x.path, x.provider)
                            for x in self.config_search_path.config_search_path
                        ]
                    ),
                ),
            )
E           hydra.errors.MissingConfigException: Cannot find primary config file: config.yml
E           Search path:
E               pkg://hydra.conf (from hydra)
E               pkg://hydra.test_utils.configs (from test)
```

File: recipes/hydra/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% set name = "hydra" %}
```
Comment (chrisburr) [inline recipes/hydra/meta.yaml]: As there are quite a lot of packages called "hydra" would this be more suitable?
```suggestion
{% set name = "facebook-hydra" %}
```

Comment (scopatz) [reply recipes/hydra/meta.yaml]: I don't know that this is a blocker for me

Comment (mdraw) [reply recipes/hydra/meta.yaml]: If a name change is required, I would rather suggest something more related to its purpose, like "hydra-config".
@omry, do you have any opinion on this? By the way, should I add you as a feedstock maintainer and can you help with the CI issues?

Comment (omry) [reply recipes/hydra/meta.yaml]: > If a name change is required, I would rather suggest something more related to its purpose, like "hydra-config".
> @omry, do you have any opinion on this? By the way, should I add you as a feedstock maintainer and can you help with the CI issues?

hydra-core, to match the pip package name (same problem in pypi).

I have no experience maintaining conda packages.
Happy to help with any issues as long as you own the process.
I can also update the documentation to reflect that conda install is a supported installation method once this land.

Comment (omry) [reply recipes/hydra/meta.yaml]: @mdraw, can you join the Hydra chat? we can talk about the failure.

File: recipes/hydra/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  requires:
+    - pytest
+  source_files:
+    - tests
+    - pytest.ini
+    - ATTRIBUTION
+    - examples
+    - hydra/test_utils
+  commands:
+    - pytest -k 'not conf_loader'  # conf_loader test doesn't find config.yml
```
Comment (omry) [inline recipes/hydra/meta.yaml]: the conf loader is a critical part of Hydra. it not working is suggesting something is wrong.

**Comment by omry** [comment]
Any advice on how to reproduce those problems locally?
I have never built a conda package (let alone on Windows).

**Comment by scopatz** [comment]
This is a noarch: python recipe so it only needs to run on Linux

**Comment by omry** [comment]
Hydra officially support Windows as well. I think it's good to test it on Windows. (My CI does cover Windows).

**Comment by scopatz** [comment]
This isn't a failure with hydra on windows, it is a known failure with building noarch packages on windows.  The packages themselves will still work perfectly well on windows

**Comment by hadim** [comment]
Any progress here? Should I submit a new PR?

**Comment by omry** [comment]
@mdraw, do you intend to complete this?

**Comment by mdraw** [comment]
Sorry, I've been busy with other work. I'll try getting it to work this weekend. @hadim do you have any specific suggestions for the PR?

**Comment by hadim** [comment]
No suggestion. Just wondering if it was going to be merge or not :-D

**Comment by omry** [comment]
I think what @mdraw is saying is that feedback and advice are welcome :)

**Comment by mdraw** [comment]
It's fixed now :) I just needed to include a few additional files in the test sources. I've also renamed the package to hydra-core. Is it ready to be merged?

**Comment by omry** [comment]
Looks good to me.
A few things:
1. System problem with appveyor on 2.7?
2. The next version of Hydra (1.0) will drop Python 2.7 support , you should be able to handle it right?

File: recipes/hydra-core/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - omegaconf <=2.0
```
Comment (omry) [inline recipes/hydra-core/meta.yaml]: Please make it as [specific](https://github.com/facebookresearch/hydra/blob/0.11_branch/setup.py#L107) as Hydra.
The intention behind that is to limit to OmegaConf 1.4.*.

**Comment by mdraw** [comment]
1. Seems to be unrelated to the recipe. I don't know why that was triggered.
2. Yes, Python 2.7 can be easily dropped in the recipe when required.

**Comment by scopatz** [comment]
Thanks!

**Comment by omry** [comment]
Awesome. How does one install hydra-core as a conda package now?

**Comment by scopatz** [comment]
You'll have to wait an hour or so and then you can do `conda install -c conda-forge hydra-core`

**Comment by omry** [comment]
Six is a dependency of OmegaConf 1.4.1 that seems to be missing (this dependency is will go away in the next version, which may have caused this confusion):

```
$ conda create -n hydra-conda python=3.6 -qy
$ conda activate hydra-conda
$ conda install hydra-core -c conda-forge -qy
$ python examples/tutorial/5_composition/my_app.py 
Traceback (most recent call last):
  File "examples/tutorial/5_composition/my_app.py", line 2, in <module>
    from omegaconf import DictConfig
  File "/home/omry/miniconda3/envs/hydra-conda/lib/python3.6/site-packages/omegaconf/__init__.py", line 2, in <module>
    from .config import Config
  File "/home/omry/miniconda3/envs/hydra-conda/lib/python3.6/site-packages/omegaconf/config.py", line 8, in <module>
    import six
ModuleNotFoundError: No module named 'six'
```

**Comment by mdraw** [comment]
Thanks @scopatz and @omry!
I don't get why that missing dependency hasn't caused test failures for the omegaconf conda package, but I'll  add the missing six to its dependencies.

**Comment by omry** [comment]
Yes, this is very odd indeed. maybe something else is installing it in the test?

[Here](https://github.com/omry/omegaconf/blob/1.4_branch/setup.py#L54) are the dependencies for 1.4.

Looking at the [deps](https://github.com/conda-forge/staged-recipes/pull/10534/files#diff-13805baf535c61a68bd5e25d7f777497R22-R25):
You also seem to add typing_extensions which is a new dependency in the next version and is required for 1.4.1.

**Comment by mdraw** [comment]
Should be fixed by https://github.com/conda-forge/omegaconf-feedstock/pull/1.

**Comment by mdraw** [comment]
Oops, I saw the typing_extensions comment too late. Yes, it seems like I had mistakenly looked at the dependencies of the master branch.

**Comment by scopatz** [comment]
Please fix it in the feedstock ;)

**Comment by mdraw** [comment]
I've already committed a fix, but Github currently errors (errorcode 50p) when I try submitting a PR, so I'll try again in a few hours.

**Comment by mdraw** [comment]
I've already committed a fix, but Github currently errors (errorcode 500)
when I try submitting a PR, so I'll try again in a few hours.

**Comment by mdraw** [comment]
Github works again, and the typing-extensions requirement is now removed from the omegaconf package.

**Pull request merged**